src/main/scala/devices/uart/UARTRx.scala

```diff
@@ -76,11 +84,23 @@ class UARTRx(c: UARTParams) extends Module {
         sample_count := countdown(c.oversample-1, 0)
 
         when (sample_mid) {
-          when (data_last) {
-            state := s_idle
-            valid := Bool(true)
-          } .otherwise {
-            shifter := Cat(voter, shifter >> 1)
+          if (c.includeParity) {
```

Comment:
Some comments about what logic is doing would be useful for maintainability going forward!